### ISSUE TITLE
Improve Placeholder instructions accessibility

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42738]
### Description

Follow-up to https://github.com/WordPress/gutenberg/pull/38366

To make screen readers read out the Placeholder `instructions` text, https://github.com/WordPress/gutenberg/pull/38366 changed the markup to wrap the contents in a `fieldset`, and the instructions in a `legend`. There are a few problems with this approach:

- it's semantically incorrect. The `legend` element is meant to give the `fieldset` an accessible name. It's not a place for generic instructions. Instead, it's meant to establish an association between the fieldset name and the control labels, to clarify the labels. A good explanation of correct usage is available on the GOV.UK blog at https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/
- More pragmatically, some screen readers repeat the legend text when focusing each control contained within the fieldset. This adds a lot of noise for screen reader users.
- Worth also noting that sometimes the instructions text is so generic that is a bit pointless. For example: 'Insert a table for sharing data' doesn't help users to understand how to use the block controls nor clarifies the control labels.

I do understand the good intent of the change from https://github.com/WordPress/gutenberg/pull/38366 as it's important to make sure important instructions are available to all users. However, I'd like to propose to explore a better solution.

Lastly, sometimes the Placeholder doesn't render any control but it still renders a fieldset and legend. That's basically a semantic `group` that doesn't actually group anything.

See screenshots below to illustrate some cases, tested with VoiceOver and Safari.

### Step-by-step reproduction instructions

- Use Safari and VoiceOver.
- Add blocks with a placeholder e.g.: Image, Columns, Table.
- Observe that when the block is added for the first time, the instructions text is read out twice.
- Press Tab to navigate to the other controls within the Placeholder.
- Observe that the instructions text is read out when focusing each control.

### Screenshots, screen recording, code snippet

Instructions read out twice when adding a block for the first time:

<img width="770" alt="first insertion twice" src="https://user-images.githubusercontent.com/1682452/181262194-898a022c-54a0-4af3-9fc8-8fc0a3e5a0ae.png">

Table block: the Instructions text 'Insert a table for sharing data.' is repeated for each control, which is redundant and a bit pointless:

<img width="1032" alt="placeholder legend 01" src="https://user-images.githubusercontent.com/1682452/181262683-a4ea17c2-ad86-4180-bb71-cb1daa39a957.png">

Audio block: the Instructions text 'Upload an audio file, pick one from your media library, or add one with a URL.' is repeated for each control:

<img width="1058" alt="placeholder legend 02" src="https://user-images.githubusercontent.com/1682452/181263115-89c36378-314b-477f-baa7-9b0b1f6594f3.png">

Columns block (BlockVariationPicker): the Instructions text 'Select a variation to start with.'is repeated for each control:

<img width="954" alt="placeholder legend 03" src="https://user-images.githubusercontent.com/1682452/181263981-c240a77b-c3fe-4ae9-b3b7-7c1912b37f5d.png">

Template part block: the Instructions text 'Choose an existing template part or create a new one.' is repeated for each control:

<img width="1046" alt="placeholder legend 04" src="https://user-images.githubusercontent.com/1682452/181264396-61b7d1af-b5d8-477b-9aee-30999f5e0d2f.png">

Table of Contents block: the fieldset is empty: it shouldn't be a fieldset in the first place:

<img width="1431" alt="placeholder empty fieldset table of contents" src="https://user-images.githubusercontent.com/1682452/181264848-bd351bb6-e16c-4c5b-814f-8c5d12fe2759.png">

Various media block: when the user doesn't have permission to upload media, the fieldset is empty and the legend is only used to show the text 'To edit this block, you need permission to upload media.'. This should be a notice rather than a fieldset + legend:

<img width="1021" alt="placeholder empty fieldset media and text no upload permission" src="https://user-images.githubusercontent.com/1682452/181265527-07364628-54ee-4564-a8aa-63580ce493be.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Thinking a bit more in depth, I'm not sure the Placeholder can render a `fieldset` + `legend` element in the first place. That would only be appropriate when the placeholder content is made of form controls. However, there are cases where the content is not made of form controls. Some examples (there are more ones):

Latest Posts block. When there are no published posts, the informative text is rendered within a fieldset:

<img width="680" alt="latest posts no posts found" src="https://user-images.githubusercontent.com/1682452/201949705-976e9798-68e8-4eee-a1ae-533302828b6d.png">

Calendar block. When there are no published posts, the informative  text is rendered within a fieldset:
 
<img width="686" alt="calendar no posts yet" src="https://user-images.githubusercontent.com/1682452/201949952-03f1389f-2a9a-40f3-b330-f5b7b9eb8673.png">

Categories list and Reusable blocks. While the block is 'resolving', ie. still fetching content, the fieldset only contains a spinner:

<img width="707" alt="reusable block resolving spinner" src="https://user-images.githubusercontent.com/1682452/201950528-dc27c858-33ef-4f7a-8f07-6e2de0b52418.png">

<img width="683" alt="categories list resolving" src="https://user-images.githubusercontent.com/1682452/201950530-a4cc4afa-dcde-4514-8593-0f885fd30cc6.png">

More importantly: to my understanding third party blocks could provide placeholder with any kind of content, even with nested forms and fieldsets, which is less than ideal. As there's no way to predict what kind of content will be within the placeholder, assuming a fieldset is hte most appropriate markup seems not ideal.
